extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js

```diff
@@ -438,12 +406,44 @@ class LifecycleBucketProcessor {
         });
     }
 
+    _initSTSClient() {
+        if (this._lcConfig.auth.type === authTypeVault) {
+            const credentials = this.credentialsManager.getCredentials({
+                id: 'lifecycle:bucketProcessor:stsclient',
+                authConfig: this._lcConfig.auth.vault.serviceCredentials,
+            });
+
+            this._stsclient = createSTSClient({
+                transport: this._lcConfig.auth.transport,
+                host: this._lcConfig.auth.vault.host,
+                port: this._lcConfig.auth.vault.port,
+                agent: new http.Agent({ keepAlive: true }),
```

Comment:
should this agent be http or https based on the transport?

Reply:
this will be cleaned up after I have broken the PR into smaller PRs. still have some things to iron out